src/ui/wxWidgets/PWStree.cpp

```diff
@@ -819,3 +819,135 @@ void PWSTreeCtrl::SetFilterState(bool state)
   if (root)
     ColourChildren(this, root, *colour);
 }
+
+/**
+ * Saves the state of all groups that have child items in tree view.
+ */
+void PWSTreeCtrl::SaveGroupDisplayState()
+{
+  auto groupstates = GetGroupDisplayState();
+  
+  if (!groupstates.empty()) {
+    m_core.SetDisplayStatus(groupstates);
+  }
+}
+
+/**
+ * Restores the state of each individual group in the tree.
+ * If the amount of groups in the tree view differs from the 
+ * amount of stored group states nothing will be done.
+ */
+void PWSTreeCtrl::RestoreGroupDisplayState()
+{
+  auto currentstates = GetGroupDisplayState();
+  auto groupstates   = m_core.GetDisplayStatus();
+  
+  if (currentstates.size() != groupstates.size()) {
+    return;
+  }
+  
+  if (!groupstates.empty()) {
+    SetGroupDisplayState(groupstates);
+  }
+}
+ 
+/**
+ * Collects the state of all group related items.
+ * 
+ * A group can be in state expanded or collapsed. The state for an expanded group item 
+ * is represented as <i>true</i>, whereas a collapsed group is reflected via <i>false</i>.
+ * 
+ * @note The booleans in the vector from first to last index represent the groups as
+ *       they appear in the tree from top to bottom.
+ * @return a vector of booleans representing the state of each group.
+ */
+std::vector<bool> PWSTreeCtrl::GetGroupDisplayState()
+{
+  std::vector<bool> groupstates;
+  
+  TraverseTree(
+    GetRootItem(), 
+    [&]
+    (wxTreeItemId itemId) -> void { 
+      groupstates.push_back(IsExpanded(itemId));
+    }
+  );
+  
+  return groupstates;
+}
+
+/**
+ * Sets the state of each individual group to be visually expanded or collapsed.
+ * 
+ * A boolean of <i>true</i> in the vector will lead to an expanded group, whereas 
+ * a boolean of <i>false</i> will trigger collapsing of a group.
+ * 
+ * @note The booleans in the vector from first to last index represent the groups as
+ *       they appear in the tree from top to bottom.
+ * @param groupstates a vector of booleans representing the state of each group.
+ */
+void PWSTreeCtrl::SetGroupDisplayState(const std::vector<bool> &groupstates)
+{
+  int groupIndex = 0;
+  
+  TraverseTree(
+    GetRootItem(), 
+    [&]
+    (wxTreeItemId itemId) -> void { 
+      if (groupIndex < groupstates.size())
+        groupstates[groupIndex++] ? Expand(itemId) : Collapse(itemId);
+    }
+  );
+}
+
+/**
+ * Sets the state for each individual group item to be visually expanded.
+ */
+void PWSTreeCtrl::SetGroupDisplayStateAllExpanded()
+{
+  auto groupstates = GetGroupDisplayState();
+  
+  for (auto &&state : groupstates) {
+    state = true;
+  }
+  
+  SetGroupDisplayState(groupstates);
+}
+
+/**
+ * Sets the state for each individual group item to be visually collapsed.
+ */
+void PWSTreeCtrl::SetGroupDisplayStateAllCollapsed()
```

Comment:
Indeed, it could be merged, but in opinion the code is more expressive and easier to read with  SetGroupDisplayStateAllCollapsed() and SetGroupDisplayStateAllExpanded() than SetGroupDisplayStateForAll(true) and SetGroupDisplayStateForAll(false), where it needs to be looked up what this boolean means. In the end, it's a matter of taste, I guess, but the current version doesn't add a lot of additional bytes to the resulting binary.

Reply:
I see your point. Too bad C++ doesn't have named arguments.